src/Providers/FakeProvider.php

```diff
@@ -0,0 +1,41 @@
+<?php
+
+namespace Worksome\CdnHeaders\Providers;
+
+use Worksome\CdnHeaders\Contracts\CdnHeadersProvider;
+
+class FakeProvider implements CdnHeadersProvider
+{
+    public function __construct(private array $config)
+    {
+    }
+
+    public function getCountryCode(): ?string
+    {
+        return $this->config['default-country'] ?? null;
+    }
+
+    public function hasCountryCode(): bool
+    {
+        return (bool) $this->getCountryCode();
+    }
+
+    public function getConnectingIp(): ?string
+    {
+        return null;
+    }
+
+    /**
+     * https://support.cloudflare.com/hc/en-us/articles/203118044-What-is-the-CF-RAY-header-#h_f7a7396f-ec41-4c52-abf5-a110cadaca7c
+     *
+     * @return non-empty-string|null
+     */
+    public function getTraceId(): ?string
```

Comment:
I think we prefer this syntax for nullabl types (can update this in a few other places too). 👍🏻
```suggestion
    public function getTraceId(): string|null
```